src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -222,5 +227,31 @@ private void DisposeBuffers()
             ((IDisposable)CurrentRenderTree).Dispose();
             _latestDirectParametersSnapshot?.Dispose();
         }
+
+        internal Task DisposeInBatchAsync(RenderBatchBuilder batchBuilder)
+        {
+            _componentWasDisposed = true;
+
+            CleanupComponentStateResources(batchBuilder);
```

Comment:
Is the order of calling `CleanupComponentStateResources(batchBuilder);` before `DisposeAsync` important? Because in `TryDisposeInBatch` it is called after `Dispose`.

Reply:
It is not important, but it makes a bit more sense in the async case to do it before instead of after since dispose async can take longer and you don't want to be holding on to resources you aren't going to need. I was conservative and didn't change the behavior on `TryDisposeInBatch` just in case. Mostly because I don't think it matters in that case.